### PR TITLE
:sparkles: Source Facebook Marketing: remove testing deleted records from integration test, move it to mock server tests

### PR DESCRIPTION
**Pull request by darynaishchenko**
resolved: https://github.com/airbytehq/airbyte-internal-issues/issues/6552

What:
removed integration tests that were based on live data and added new test suit for deleted records to  mock server tests (unit_tests/integration)

### CODE REVIEW

File: airbyte-integrations/connectors/source-facebook-marketing/unit_tests/integration/response_builder.py [hunk 1]
```diff
@@ -31,3 +34,19 @@ def error_reduce_amount_of_data_response() -> HttpResponse:
         "error": {"code": 1, "message": "Please reduce the amount of data you're asking for, then retry your request"},
     }
     return build_response(body=response, status_code=HTTPStatus.INTERNAL_SERVER_ERROR)
+
+
+def get_ads_response() -> HttpResponse:
```
Comment (askarpets) [inline airbyte-integrations/connectors/source-facebook-marketing/unit_tests/integration/response_builder.py:39]: Should `create_response_builder` be used here?

File: airbyte-integrations/connectors/source-facebook-marketing/unit_tests/integration/test_include_deleted.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+            "tracking_specs",
+            "conversion_specs",
+        ]
+
+        http_mocker.get(
+            get_ads_request().with_limit(100).with_filtering(filters).with_fields(fields).with_summary().build(),
+            get_ads_response(),
+        )
+
+        output = self._read(config().with_ad_statuses(self.statuses), "ads")
```
Comment (askarpets) [inline airbyte-integrations/connectors/source-facebook-marketing/unit_tests/integration/test_include_deleted.py:67]: With `create_response_builder`  added, `with_record` could be applied here so it is more clear why one record is expected

File: airbyte-integrations/connectors/source-facebook-marketing/unit_tests/resource/http/response/ad_sets.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{
```
Comment (askarpets) [inline airbyte-integrations/connectors/source-facebook-marketing/unit_tests/resource/http/response/ad_sets.json:1]: Could you please update these files so the data looks exactly like real responses?

**Review by askarpets — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-facebook-marketing/unit_tests/integration/response_builder.py (hunk 1 above)
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-facebook-marketing/unit_tests/integration/response_builder.py]: updated

File: airbyte-integrations/connectors/source-facebook-marketing/unit_tests/integration/test_include_deleted.py (hunk 2 above)
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-facebook-marketing/unit_tests/integration/test_include_deleted.py:90]: updated

File: airbyte-integrations/connectors/source-facebook-marketing/unit_tests/resource/http/response/ad_sets.json (hunk 3 above)
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-facebook-marketing/unit_tests/resource/http/response/ad_sets.json:1]: updated, we have these records in expected records file

**Review by askarpets — APPROVED** [review]

File: airbyte-integrations/connectors/source-facebook-marketing/unit_tests/integration/test_include_deleted.py [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+
+    @HttpMocker()
+    def test_ads_stream(self, http_mocker: HttpMocker):
+        http_mocker.get(
+            get_account_request().build(),
+            get_account_response(),
+        )
+        filters = (
+            '[{"field":"ad.effective_status","operator":"IN","value":["ACTIVE","ARCHIVED"]},'
+            '{"field":"ad.updated_time","operator":"GREATER_THAN","value":1672531200}]'
```
Comment (maxi297) [inline airbyte-integrations/connectors/source-facebook-marketing/unit_tests/integration/test_include_deleted.py:60]: Where does this value comes from? It seems like it should be relevant but without a variable, it is hard to know the link between this and where it comes from

File: airbyte-integrations/connectors/source-facebook-marketing/unit_tests/integration/test_include_deleted.py [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+
+        http_mocker.get(
+            get_ads_request().with_limit(100).with_filtering(filters).with_fields(fields).with_summary().build(),
+            _stream_response("ads").with_record(_stream_record("ads")).build(),
+        )
+
+        output = self._read(config().with_ad_statuses(self.statuses), "ads")
+        assert len(output.records) == 1
+        account_state = output.most_recent_state.dict()["stream_state"][self.account_id]
+        assert self.filter_statuses_flag in account_state, f"State should include `filter_statuses` flag"
```
Comment (maxi297) [inline airbyte-integrations/connectors/source-facebook-marketing/unit_tests/integration/test_include_deleted.py:93]: The error message is clear but without context about the "why", it is hard to understand why this is asserted here. Can we add more info on that?

File: airbyte-integrations/connectors/source-facebook-marketing/unit_tests/integration/test_include_deleted.py [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+        )
+
+    @HttpMocker()
+    def test_ads_stream(self, http_mocker: HttpMocker):
+        http_mocker.get(
+            get_account_request().build(),
+            get_account_response(),
+        )
+        filters = (
+            '[{"field":"ad.effective_status","operator":"IN","value":["ACTIVE","ARCHIVED"]},'
```
Comment (maxi297) [inline airbyte-integrations/connectors/source-facebook-marketing/unit_tests/integration/test_include_deleted.py:59]: Should we use `self. statuses` here to clarify the link between the config and the API call? The same comment applies for the other tests

File: airbyte-integrations/connectors/source-facebook-marketing/unit_tests/integration/test_include_deleted.py [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+        http_mocker.get(
+            get_ads_request().with_limit(100).with_filtering(filters).with_fields(fields).with_summary().build(),
+            _stream_response("ads").with_record(_stream_record("ads")).build(),
+        )
+
+        output = self._read(config().with_ad_statuses(self.statuses), "ads")
+        assert len(output.records) == 1
+        account_state = output.most_recent_state.dict()["stream_state"][self.account_id]
+        assert self.filter_statuses_flag in account_state, f"State should include `filter_statuses` flag"
+        assert account_state == {"filter_statuses": ["ACTIVE", "ARCHIVED"], "updated_time": "2023-03-21T22:41:46-0700"}
```
Comment (maxi297) [inline airbyte-integrations/connectors/source-facebook-marketing/unit_tests/integration/test_include_deleted.py:94]: Should we use `self. statuses` here?

**Review by maxi297 — APPROVED** [review]
I haven't check if the coverage between the integration tests and those were equivalent but it seems like a good start. For example, were the integration tests implicitly covering pagination? Assuming the coverage is the same, I'm good with this change

**Review by bazarnov — APPROVED** [review]

File: airbyte-integrations/connectors/source-facebook-marketing/unit_tests/integration/test_include_deleted.py (hunk 7 above)
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-facebook-marketing/unit_tests/integration/test_include_deleted.py:95]: updated with self.statuses

File: airbyte-integrations/connectors/source-facebook-marketing/unit_tests/integration/test_include_deleted.py (hunk 6 above)
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-facebook-marketing/unit_tests/integration/test_include_deleted.py]: updated with self.statuses

File: airbyte-integrations/connectors/source-facebook-marketing/unit_tests/integration/test_include_deleted.py (hunk 5 above)
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-facebook-marketing/unit_tests/integration/test_include_deleted.py]: updated error message

File: airbyte-integrations/connectors/source-facebook-marketing/unit_tests/integration/test_include_deleted.py (hunk 4 above)
Comment (darynaishchenko) [reply airbyte-integrations/connectors/source-facebook-marketing/unit_tests/integration/test_include_deleted.py]: added comment for filters

**Pull request merged**